### PR TITLE
Remove vertical bar from some rules that convert to MM characters. Update Zawgyi conversion rules to fix issues with medials

### PR DESCRIPTION
**Pull request by LinKyaw**
Z -> U convert Vertical bar "|" to myanmar characters( ျ,ှ)

U -> Z
    expected "အ‌ေႂကြး" -> actual "အ‌ႂကြေး".
    expected "ေကြၽးေမြး" -> actual "က်ြေးေမြး".
    expected "ဆႏၵႀကီး" -> actual "ဆႁႏၵကီး"
    changes to use correct medials.

### CODE REVIEW

File: clients/java/src/test/java/com/google/myanmartools/TransliterateTest.java [hunk 1]
```diff
@@ -568,7 +568,7 @@
           "ဝေသစ်လွင် about"
           },
     {  // #101 From case #134: Other
-      "ျပန္လမ္းမဲ့ကၽြန္းအပိုင္း၄",
+      "ျပန္လမ္းမဲ့ကြၽန္းအပိုင္း၄",
```
Comment (sven-oly) [inline clients/java/src/test/java/com/google/myanmartools/TransliterateTest.java]: I'm not sure why this test case is being changed. The old and new Z forms appear identical. Both should convert to the same Unicode string. Perhaps the test should include the two versions of the Z input, e.g, 
 "ျပန္လမ္းမဲ့ကၽြန္းအပိုင္း၄" +  "ျပန္လမ္းမဲ့ကြၽန္းအပိုင္း၄",
with the same Unicode string twice for expected output.

File: clients/js/spec/zawgyi_converter_spec.js [hunk 2]
```diff
@@ -546,7 +546,7 @@ var zawygi_unicode_convert_data = [
       u: "ဝေသစ်လွင် about"
     },
     {  // #101 From case #134: Other
-      z: "ျပန္လမ္းမဲ့ကၽြန္းအပိုင္း၄",
+      z: "ျပန္လမ္းမဲ့ကြၽန္းအပိုင္း၄",
```
Comment (sven-oly) [inline clients/js/spec/zawgyi_converter_spec.js]: Again, why change the Z test string? Also, please add the same test strings as in the Java test file.

File: genconvert/input/my-t-my-s0-zawgyi.txt [hunk 3]
```diff
@@ -63,7 +63,7 @@ $wspace = [\u0020\u00a0\u1680\u2000-\u200d\u2060\u202f\u205f\u3000\ufeff];
 \u1086 → \u103F ;
 
 # yapin
-[\u103A|\u107d] → \u103B ;
+[\u103A\u107d] → \u103B ;
```
Comment (sven-oly) [inline genconvert/input/my-t-my-s0-zawgyi.txt]: Thanks for fixing these two rules.

File: genconvert/input/my-t-my-d0-zawgyi.txt [hunk 4]
```diff
@@ -36,8 +36,8 @@ $vowelmedial = [\u102B-\u1030\u1032\u103c-\u103F];
 $ukinzi = [\u1004\u101b\u105a]\u103A\u1039;
 
 $medialraZ = [\u103b\u107e-\u1084];
-$lowsignZ = [\u102f\u1030\u1037\u103a\u103c\u103d\u1087-\u108a];
```
Comment (sven-oly) [inline genconvert/input/my-t-my-d0-zawgyi.txt]: Z-103a has a descender that should be considered in picking the medial ra code. I think this should stay in lowsignZ.

File: genconvert/input/my-t-my-d0-zawgyi.txt [hunk 5]
```diff
@@ -36,8 +36,8 @@ $vowelmedial = [\u102B-\u1030\u1032\u103c-\u103F];
...
-$highsignZ = [\u102d\u102e\u1032\u1036\u1039\u103d-\u103e\u1064];
```
Comment (sven-oly) [inline genconvert/input/my-t-my-d0-zawgyi.txt]: Good catch!

File: genconvert/input/my-t-my-s0-zawgyi.txt [hunk 6]
```diff
@@ -73,7 +73,7 @@ $wspace = [\u0020\u00a0\u1680\u2000-\u200d\u2060\u202f\u205f\u3000\ufeff];
 \u103C → \u103D ;
 
 # hatoh
-[\u103D|\u1087] → \u103E ;
+[\u103D\u1087] → \u103E ;
```
Comment (sven-oly) [inline genconvert/input/my-t-my-s0-zawgyi.txt]: Good catch!

File: genconvert/input/my-t-my-d0-zawgyi.txt [hunk 7]
```diff
@@ -186,18 +187,20 @@ $ukinzi ($consonant) > $1 \u1064 ;
 ## Phase 2: Further adjustments
 ::Null;
 
-# Handle consonant, subscripted consonant, medial ra
-($narrowconsonant) ($subscriptitem) ($highsignZ) $medialraZ > \u1083 $1 $2 $3 ;
-($wideconsonant) ($subscriptitem) ($highsignZ) $medialraZ > \u1084 $1 $2 ;
-($narrowconsonant) ($subscriptitem) $medialraZ > \u1081 $1 $2 ;
-($wideconsonant) ($subscriptitem) $medialraZ > \u1082 $1 $2 ;
+# Two medials
+\u103a \u103c > \u103c \u107d;
```
Comment (sven-oly) [inline genconvert/input/my-t-my-d0-zawgyi.txt]: Do you have a test case for this?

File: clients/java/src/test/java/com/google/myanmartools/TransliterateTest.java (hunk 1 above)
Comment (LinKyaw) [reply clients/java/src/test/java/com/google/myanmartools/TransliterateTest.java]: yeah, both Z strings convert to the same Unicode string. But Unicode string also convert to only one Z string. In this case, "ျပန္လမ္းမဲ့ကြၽန္းအပိုင္း၄" should be the right output. if there are two versions of the Z input, the test will fail.

File: clients/js/spec/zawgyi_converter_spec.js (hunk 2 above)
Comment (LinKyaw) [reply clients/js/spec/zawgyi_converter_spec.js]: Same case as in Java. I will add the same test strings as in the Java test file.

File: genconvert/input/my-t-my-d0-zawgyi.txt (hunk 4 above)
Comment (LinKyaw) [reply genconvert/input/my-t-my-d0-zawgyi.txt]: Actually Z-103A is also the ra with different usage. There is no usage combining those two.

File: genconvert/input/my-t-my-d0-zawgyi.txt (hunk 7 above)
Comment (LinKyaw) [reply genconvert/input/my-t-my-d0-zawgyi.txt]: Yes, case 114 and 115 are for that and also 101.

**Comment by LinKyaw** [comment]
Z test string need to change because as in issue #21
Input: U "**1000 103b 103d** 1014 103a 1038"
expected: Z "**1000 103c 107d** 1014 1039 1038 "
actual: Z "**1000 107d 103c** 1014 1039 1038"

this does solve the issue #21 U->Z error. And for that the test strings are:
Z "ျပန္လမ္းမဲ့ကြၽန္းအပိုင္း၄" <-> U "ပြန်လမ်းမဲ့ကျွန်းအပိုင်း၎" 
Z "ေကြၽးေမြး" <-> U "ကျွေးမွေး" // before result "က်ြေးေမြး"

this also solve for
input: U "ဆန္ဒကြီး ကြွေးကြော်"
expected: Z "ဆႏၵႀကီး ေႂကြးေၾကာ္"
actual: Z "ဆႁႏၵကီး ႂကြေးေၾကာ္"

As for issue #21 Z->U, I consider "108f 103c 1036 1031 1010 **1038 103a**" as an typo.
Z-1038 should not come before U-103a. we don't type "**1038 103a**". It's not even correct in Zawgyi form.

File: clients/java/src/test/java/com/google/myanmartools/TransliterateTest.java (hunk 1 above)
Comment (sven-oly) [reply clients/java/src/test/java/com/google/myanmartools/TransliterateTest.java]: Yes, this makes sense. Thx.

**Comment by sffc** [comment]
@theinlinkyaw Hi there!  We realized that this change to Myanmar Tools was never integrated upstream into Unicode CLDR.  Would you mind giving us permission to add your work to Unicode CLDR by signing the Unicode CLA?

https://cla-assistant.io/unicode-org/cldr

Thank you!

**Pull request merged**